hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestCOWDataSource.scala

```diff
@@ -1494,15 +1494,11 @@ class TestCOWDataSource extends HoodieSparkClientTestBase with ScalaAssertionSup
 
     assert(firstDF.count() == 2)
 
-    // data_date is the partition field. Persist to the parquet file using the origin values, and read it.
-    // TODO(HUDI-3204) we have to revert this to pre-existing behavior from 0.10
```

Comment:
HUDI-3204 is what this PR also fixes.  I see other `TODO(HUDI-3204)` that are not addressed by this PR.  Could you check and fix all of them?

Reply:
How hard will it be to fix the existing code path (0.x, 1.x with file group reader disabled)?

Reply:
We can create a separate ticket to track the fix in the fallback code path.

Reply:
https://issues.apache.org/jira/browse/HUDI-8168